src/storage/memory_store.ts

```diff
@@ -0,0 +1,821 @@
+   *
+   * @async
+   * @returns {AsyncGenerator<ITokenData & Partial<ITokenMetadata>>}
+   */
+  async *tokenIter(): AsyncGenerator<ITokenData & Partial<ITokenMetadata>> {
+    for (const tokenInfo of this.tokens.values()) {
+      const tokenMeta = this.tokensMetadata.get(tokenInfo.uid);
+      yield { ...tokenInfo, ...tokenMeta };
+    }
+  }
+
+  /**
+   * Get a token on storage from the uid
+   * @param tokenUid The token id to fetch
+   * @returns {Promise<(ITokenData & Partial<ITokenMetadata>) | null>} The token data if present
+   */
+  async getToken(tokenUid: string): Promise<(ITokenData & Partial<ITokenMetadata>) | null> {
+    const tokenConfig = this.tokens.get(tokenUid);
+    if (tokenConfig === undefined) {
+      return null;
+    }
+    const DEFAULT_TOKEN_META: ITokenMetadata = {
+      numTransactions: 0,
+      balance: {
+        tokens: { unlocked: 0, locked: 0 },
+        authorities: {
+          mint: { unlocked: 0, locked: 0 },
+          melt: { unlocked: 0, locked: 0 },
+        },
+      }
+    };
+    const tokenMeta = this.tokensMetadata.get(tokenUid);
+
+    return { ...tokenConfig, ...DEFAULT_TOKEN_META, ...tokenMeta };
+  }
+
+  /**
+   * Save a token on storage
+   * @param {ITokenData} tokenConfig Token config
+   * @param {ITokenMetadata|undefined} [meta] The token metadata
+   * @async
+   * @returns {Promise<void>}
+   */
+  async saveToken(tokenConfig: ITokenData, meta?: ITokenMetadata | undefined): Promise<void> {
+    if (this.tokens.has(tokenConfig.uid)) {
+      throw new Error('Already have this token');
+    }
+    this.tokens.set(tokenConfig.uid, tokenConfig);
+    if (meta !== undefined) {
+      this.tokensMetadata.set(tokenConfig.uid, meta);
+    }
+  }
+
+  /**
+   * Iterate on registered tokens.
+   *
+   * @async
+   * @returns {AsyncGenerator<ITokenData & Partial<ITokenMetadata>>}
+   */
+  async *registeredTokenIter(): AsyncGenerator<ITokenData & Partial<ITokenMetadata>> {
+    for (const tokenConfig of this.registeredTokens.values()) {
+      const tokenMeta = this.tokensMetadata.get(tokenConfig.uid);
+      yield { ...tokenConfig, ...tokenMeta };
+    }
+  }
+
+  /**
+   * Register a token.
+   *
+   * Obs: we require the token data because the token being registered may not be on our storage yet.
+   *
+   * @param token Token config to register
+   * @async
+   * @returns {Promise<void>}
+   */
+  async registerToken(token: ITokenData): Promise<void> {
+    this.registeredTokens.set(token.uid, token);
+  }
+
+  /**
+   * Unregister a token.
+   *
+   * @param {string} tokenUid Token id
+   * @async
+   * @returns {Promise<void>}
+   */
+  async unregisterToken(tokenUid: string): Promise<void> {
+    this.registeredTokens.delete(tokenUid);
+  }
+
+  /**
+   * Delete tokens from storage.
+   * @param tokens A list of token ids to delete
+   * @async
+   * @returns {Promise<void>}
+   */
+  async deleteTokens(tokens: string[]): Promise<void> {
+    for (const tokenUid of tokens) {
```

Comment:
This was somewhat confusing since the methods to manage tokens on storage were present on the old token utils but I did not find anywhere that uses them,
I translated to be a reasonable alternative, but if a token enters your history, it should only be deleted if the transaction that adds it was voided, which would be solved when the history is processed.

So I don't think we should change it now, unless there is a wallet that requires this method, which I think is unlikely.

Reply:
I don't understand why we would have registered tokens from tokens that we don't even have the referece.

Reply:
The old token utilities had some token management methods, but since they are already supported with the storage methods to manage registered tokens we don't need this so it has been removed.